l10n_it_fatturapa_out_oss/wizard/wizard_export_fatturapa.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Sergio Corato <https://github.com/sergiocorato>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.addons.l10n_it_account.tools.account_tools import encode_for_export
+from odoo.tools.float_utils import float_round
+from odoo.addons.l10n_it_fatturapa.bindings.fatturapa import (
+    AltriDatiGestionaliType,
+)
+
+
+class WizardExportFatturapa(models.TransientModel):
+    _inherit = "wizard.export.fatturapa"
+
+    def setDettaglioLinea(self, line_no, line, body, price_precision, uom_precision):
+        DettaglioLinea = super().setDettaglioLinea(
+            line_no, line, body, price_precision, uom_precision)
+
+        invoice_line_tax = line.invoice_line_tax_ids[0]
+        if invoice_line_tax.oss_subjected and line.invoice_id.company_id.oss_year_ids.\
+            filtered(
+                lambda x: x.year == str(line.invoice_id.date.year) and x.oss_subjected
+                ) and invoice_line_tax.kind_id:
+            dati_gestionali = AltriDatiGestionaliType(
+                TipoDato="OSS",
+                RiferimentoTesto=encode_for_export(
+                    '%.2f' % float_round(invoice_line_tax.amount, 2), 60),
+            )
+            DettaglioLinea.AltriDatiGestionali.append(dati_gestionali)
+            AliquotaIVA = '0.00'
+            DettaglioLinea.AliquotaIVA = AliquotaIVA
+            DettaglioLinea.Natura = invoice_line_tax.kind_id.code
+        return DettaglioLinea
+
+    def setDatiRiepilogo(self, invoice, body):
+        super().setDatiRiepilogo(invoice, body)
+
+        if invoice.company_id.oss_year_ids.filtered(
```

Comment:
Come mai anche il filtro su `oss_year_ids`? Se sto generando l'XML di una fattura OSS, vorrò che sia sempre fatta in questo modo, cioè con aliquota 0 e N3.2, giusto?

Reply:
La normativa OSS è applicabile per anno in base al fatturato intra, quindi no

Reply:
Ma se alla mia azienda non si applica, allora non userò nemmeno un'imposta OSS, giusto?
Cioè, se non si applica e vendo in francia a un privato, non metto aliquota francese 20%, giusto?

Reply:
Sì esatto, si applica solo a seguito del superamento di una soglia di fatturato verso privati nella zona intra.

Reply:
Quindi il controllo sugli anni qua è superfluo, in quanto `y.oss_subjected` sarà False.

Applicare o meno le imposte OSS dipenderà dalle posizioni fiscali e la loro applicazione automatica.

Reply:
Ti propongo quindi https://github.com/sergiocorato/l10n-italy/pull/6

Poi possiamo decidere come disattivare l'applicazione delle imposte OSS in base agli anni in cui l'azienda è soggetta, ma farei un discorso a parte.

Reply:
Mmm, perchè dici di togliere una funzionalità? Non mi pare complicata da gestire.
Un modulo a parte andrebbe a dover riscrivere sulle stesse funzioni.

Reply:
La tolgo perchè mi sembra sbagliata: se delle imposte OSS sono state applicate in fattura, perchè dovrei toglierle solo dalla generazione dell'XML?
Se la mia azienda è fuori dal regime OSS, allora non avrei dovuto applicare in fattura un'imposta OSS.

Quindi la funzionalità a parte sarebbe comunque diversa: servirebbe a stabilire l'applicabilità delle imposte OSS (presumibilmente tramite posizioni fiscali) in base agli anni in cui l'azienda è soggetta, o altro.